### ISSUE TITLE
Add some new features already implemented and tested by others

### ISSUE THREAD

**Issue by JohnBetaro** [mollyim/mollyim-android#181]
### Is there an existing request for this?

- [X] I have searched the existing issues

### Feature description

  There another fork of Signal maintained by @johanw666 that has a lot of extra features listed and drtailed [Here](https://johanw.home.xs4all.nl/Signal/signal-jw.html) , you may ask why don't I use his app directly, for 2 reasons:

1- His app is built on his local machine then uploaded to GitHub releases and yours is build by GitHub Actions, so you offer more transparency than he does, and I'm not accusing him of doing anything sketchy behind the scenes, he has all my respect for his substantial effort to make this fork, I just like to see what happens in front of my eyes.

2- I like that your fork offers more security than the normal Signal app does.

3- I like the AMOLED Dark that you use more than the classic one.

  I hope you'll consider my suggestions and at least take a look to see if you can implement at least one of features listed.

**Comment by johanw666** [comment]
I've discussed this in the past with Valldrac, he has copied one feature (the map-type setting in Google maps) and disagrees with me on some other features (the disabeling of viewOnce and remote delete).

Backporting most options is certainly possible, I run currently also Molly insider linkdevice as linked device and have ported most options of my fork to that Molly instance. I probably also broke the Github actions build in the process, I didn't test that. I can send you a zipfile with all changes that you can copy over the source and then you can build yourself if you like.

**Comment by JohnBetaro** [comment]
Yeah, I read the discussion between you and it's sad that he wont implement your feature however I opened this feature request to make reconsider.

Why don't you publish that Molly fork that you made so we can have all the features ? It will be very useful and satisfying since we will get the extra security and the colors of molly, I hope you'll do it since you already built it and you're using it.

Yes I will be happy to have your zip file and try to make my own.

**Comment by johanw666** [comment]
I can't fork it anymore in Github because I already have a fork of the production Molly. It's a limitation from Github.

**Comment by JohnBetaro** [comment]
So there is no other way we can get the apk file from you ?

**Comment by johanw666** [comment]
I've made a non-connected repository of my changes. The production apk's can be found at

https://github.com/johanw666/Molly-linkdevice-JW/releases/tag/v6.18.4-1.0-JW-linkdevice

Do you also need the staging apk's? I didn't build the FOSS version and even test if building it works with my changes.

**Comment by JohnBetaro** [comment]
Thank you so much, I don't think I need the staging apks, all I need for now is molly equivalent with your features. 

Are you planning on publishing updates in the future ? 

All I need in the future is to replace to files of the latest version with the ones you gave me and I'll be done ? Should I change some things before I can make my own ?

**Comment by johanw666** [comment]
You can download the official repo (the linkdevice variant with the corresponding version number), download the zipfile and copy the files over those in the official source, load the project in Android Studio and build a release.

Of course, the signing keys from Valldrac, me and you will be different so you probably can't install such a version over a current Molly install (unless you rooted your phone and managed to disable signature checking). You need to make a backup, uninstall, reinstall and restore backup in that case.

**Comment by JohnBetaro** [comment]
But when I overwrite the files everytime won't that affect future updates ?

**Comment by johanw666** [comment]
What updates? Of the Molly repository? No, a checkout will change everything to the state in the repository on Github. But to build my changed version you need either to wait for me to make the changes or patch the files yourself with the changes in my files (warning: that isn't always enough, in the next version there will be quite some changes in the plaintext export, I had to make them already in Signal 6.19).

**Comment by JohnBetaro** [comment]
So you are planning to publish some updates in the future of the apks and the files ? This will be very gentle from you.

**Comment by johanw666** [comment]
I'll keep using the file, I will probably maintain this repository until the linkdevice changes are merged into the main Molly branch, in which case I'll post my changes in my forked Molly branch.

**Comment by JohnBetaro** [comment]
Are they planning to merge your changes or this is just a thing we hope they'll consider ?

**Comment by johanw666** [comment]
AFAIK there is one bug to fix but you'd better ask Valldrac.

**Comment by JohnBetaro** [comment]
Well, we are waiting for him to answer and the question is : Will all the features founded by @johanw666 be implemented in the future if you managed to fix the bugs ?

Thank you again johan, we will be in touch as always.

**Comment by JohnBetaro** [comment]
@johanw666 I've checked you repo branches and I found some old branches that show what did you change exactly in the code to add a specific feature.

What I didn't find is 2 main patches that I want so badly and they are the reason why I'm using your fork, "ignore remote deletion" and "treat view once media as normal media", maybe if I found out how you did implement these 2 features I can make my own fork in the future which has just what I need.

Is there anyway to see what did you change exactly in the source code just to add them without seeing the others ? Or can you create a branches that show how ?

**Comment by johanw666** [comment]
I usually put all my changes in one commit, I don't separate them. The functional changes for these methods are (since 6.28) in messages/DataMessageProcessor.kt and messages/SyncMessageProcessor.kt (the later if you don't want a desktop instance to remove the data when you view it there) plus some UI code and settings.

**Comment by JohnBetaro** [comment]
So basically these are the files I need to change to implement the 2 features I want ? Without touching something else ?

And does it also work on the normal Signal app ?

**Comment by johanw666** [comment]
No, like I said, I would need Valldracs signing key for that and that is not going to hapen.

**Comment by JohnBetaro** [comment]
I think this problem can be solved by just changing the package name, it's not a hard task to do.

**Comment by Chromatros** [comment]
Yes I was asking to change the package name

**Comment by JohnBetaro** [comment]
It's very easy if you want to it yourself, you can download apk editor app and the rest is easy, I did it many times
